api/src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -163,6 +178,22 @@ public BallColor getIconColor() {
     protected abstract String getTypeDisplayName();
 
     /**
+     * Gets a human readable text that may include a {@link StepDescriptor#getFunctionName()}.
+     * It would return "echo" for a flow node linked to an EchoStep or "ws {" for WorkspaceStep.
+     *
+     * For StepEndNode it would return "} // step.getFunctionName()".
+     *
+     * Note that this method should be abstract (supposed to be implemented in all subclasses), but keeping
+     * it non-abstract to avoid binary incompatibilities.
+     *
+     * @return the text human-readable representation of the step function name
+     *      or {@link FlowNode#getDisplayName()} by default (if not overriden in subclasses)
+     */
+    protected /* abstract */ String getTypeFunctionName() {
```

Comment:
Still seems like a poor name choice, since it is only a function name in the case of `StepAtomNode`s.